src/gui/ObsListCreateEditDialog.cpp

```diff
@@ -43,1064 +44,922 @@
-
-    // objtype (object type)
-    item = new QStandardItem(objtype);
-    item->setEditable(false);
-    obsListListModel->setItem(number, ColumnType, item);
-
-    // ra
-    item = new QStandardItem(ra);
-    item->setEditable(false);
-    obsListListModel->setItem(number, ColumnRa, item);
-
-    // dec
-    item = new QStandardItem(dec);
-    item->setEditable(false);
-    obsListListModel->setItem(number, ColumnDec, item);
-
-    // magnitude
-    item = new QStandardItem(magnitude);
-    item->setEditable(false);
-    obsListListModel->setItem(number, ColumnMagnitude, item);
-
-    // constellation
-    item = new QStandardItem(constellation);
-    item->setEditable(false);
-    obsListListModel->setItem(number, ColumnConstellation, item);
-
-    for (int i = 0; i < ColumnCount; ++i) {
-        ui->obsListCreationEditionTreeView->resizeColumnToContents(i);
-    }
+					  const QString &constellation,
+					  const QString &date,
+					  const QString &location,
+					  const QString &landscapeID)
+{
+	const int number=obsListListModel->rowCount();
+	QStandardItem *item = Q_NULLPTR;
+
+	item = new QStandardItem(olud);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnUUID, item);
+
+	item = new QStandardItem(name);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnName, item);
+
+	item = new QStandardItem(nameI18n);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnNameI18n, item);
+
+	item = new QStandardItem(objtype);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnType, item);
+
+	item = new QStandardItem(ra);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnRa, item);
+
+	item = new QStandardItem(dec);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnDec, item);
+
+	item = new QStandardItem(magnitude);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnMagnitude, item);
+
+	item = new QStandardItem(constellation);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnConstellation, item);
+
+	item = new QStandardItem(date);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnDate, item);
+
+	item = new QStandardItem(location);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnLocation, item);
+
+	item = new QStandardItem(landscapeID);
+	item->setEditable(false);
+	obsListListModel->setItem(number, ColumnLandscapeID, item);
+
+	for (int i = 0; i < ColumnCount; ++i)
+		ui->obsListCreationEditionTreeView->resizeColumnToContents(i);
 }
 
 /*
  * Slot for button obsListAddObjectButton.
  * Save selected object into the list of observed objects.
  */
-void ObsListCreateEditDialog::obsListAddObjectButtonPressed() {
-    initErrorMessage();
-    const QList<StelObjectP> &selectedObject = objectMgr->getSelectedObject();
-
-    if (!selectedObject.isEmpty()) {
-
-        // No duplicate item in the same list
-        bool is_already_in_list = false;
```

Comment:
You can delete this line, it is not useful because the variable is always false.

Reply:
I had tried to make the test more comprehensive, however it fails. OK, we accept duplicate entries for now. Maybe for 1.1...